ark/attest/__tests__/benchExpectedOutput.ts

```diff
@@ -36,7 +36,11 @@ type makeComplexType<S extends string> = S extends `${infer head}${infer tail}`
 
 bench("bench type", () => {
 	return {} as makeComplexType<"defenestration">
-}).types([169, "instantiations"])
+}).types([182, "instantiations"])
+
+bench("bench type from external module", () => {
```

Comment:
Just want to double-check that this fails before the fix since we actually have some tests that reference other packages in the repo but they passed, I assume because we have a customCondition that imports the `.ts` directly when running internally.

Reply:
Yep, I followed TDD with this one and without the change in this PR we get 0 instantiations here. 

Btw, this is the case with any external module - whether it's another local source file or one that lives in node_modules. I think the custom condition you're mentioning would explain that. If the type definition is inlined into the bench file, it works as expected without the change.

Reply:
Great! I still don't really understand the nuances of why the customCondition would not apply from `@arktype/attest` to itself but I'm glad it works out. 

I guess the last thing to verify would be that an empty bench still creates `0` instantiations. Could you add that to the template as well?

```ts
bench("empty", () => {}).types([0, "instantiations"])
```

Reply:
That's a great catch. It doesn't return 0, but 13, which is the same value the existing benches went up by. Let's hold off on merging this then and I'll investigate why this is happening tomorrow morning.

Reply:
Awesome thanks again for your diligence and all of your help!

I'm sorry for the state of the contributing experience. I hope the process will be much smoother again when I imminently merge this beta branch that is 1600 commits ahead of main 🙃

Reply:
Also want to add for context before I forget that the way the benchmarks work is by creating a temporary project including a file with only the contents of that bench alongside other top-level items in the module included.

The number of instantiations for the project without that file is subtracted from the count with the bench call contents. The first thing that comes to mind is that potentially those instantiation counts are out of sync (maybe one is still referencing an old project somehow or not being updated in the file-backed system)?

That seems especially likely now that I look at `ark/attest/bench/type.ts` and see that it is creating a virtual ts environment on line `39`, so I'd start there😊

Reply:
My guess would be the Virtual env on line 9 of `ark/attest/tsserver/tsserver.ts` should be updated to a file backed system as well then you'll be good to go.

Reply:
I tried updating that one to the file-backed system as well, but still seeing the extra 13 instantiations. 

I do suspect that those are somehow related to the program files that are being loaded [here](https://github.com/arktypeio/arktype/blob/beta/ark/attest/tsserver/tsserver.ts#L21-L27). 

If I use attest bench (with the change in this PR applied) in my own project, the empty bench actually works as expected:

![image](https://github.com/arktypeio/arktype/assets/29709626/9cc78527-50ed-472a-8e15-3d7f34c019d5)

Before I made the initial change in this PR, "bench simple schema" was returning 0 instantiations.

Reply:
@iartemiev Can you push the updated version with the `ark/attest/tsserver/tsserver.ts` changes anyways? Even if it doesn't resolve that issue it seems important they stay in sync.

I can then take a look and see if I can resolve what's going on with the extra instantiations.

Reply:
That is a crazy number of instantiations for that type by the way 😬

Reply:
Yep, will do shortly. I've got a couple more things I'd like to try before throwing in the towel. I need to run out to an appointment shortly so it may have to be in the early afternoon.

Btw, while working on this I added a ts-node debugger config for vscode (in `launch.json`) that makes it easy to debug the source using the test files as the entry point. Had to enable sourcemaps in the tsup config to get it to map back to TS. Is this something that would be useful to the project? If not, I'll take it out before I push up the next commit, otherwise, I can include it.

> That is a crazy number of instantiations for that type by the way 😬

Agreed 😅. This is a new feature we launched in preview, but it's completely un-optimized at this point. The plan is to improve that quite a bit before GA.

Reply:
I usually use the Test Explorer which is honestly amazing for this kind of thing, although it doesn't work well with `attest`'s internal tests because they require type checking.

https://github.com/arktypeio/arktype/assets/10645823/4d2a1e66-3e62-40f3-a13a-0f3985e1f03c

Not helpful in this case, but just figured I'd mention it in case you didn't know- definitely an underutilized feature IMO!

For `attest`, I normally use a JavaScript debug console and just debug the tests that way, but I'm not particularly invested one way or the other if a launch script seems more natural others might appreciate that as well (I also need to create some better contributing docs that summarize how I handle these things!).

Reply:
Makes sense. I'll strip it out just to stay consistent. I'll have to give Text Explorer a try. 

I just pushed my commit with the current state. Sorry for the force push, had to rebase `beta`.

Reply:
> Agreed 😅. This is a new feature we launched in preview, but it's completely un-optimized at this point. The plan is to improve that quite a bit before GA ([link](https://docs.amplify.aws/gen2/build-a-backend/data/set-up-data/) in case you're interested)

Ahh, looks cool! I'm a former Amazonian myself and always felt like we could be doing more in the open source ecosystem, so it's great to see projects like this.

Type optimization and runtime validation are two of my favorite things to talk about so would also be happy to follow up sometime and see if I might be able to help out!